src/Yarhl.UnitTests/IO/DataStreamTests.cs

```diff
@@ -965,6 +965,32 @@ public void ReadFormAfeterDisposeThrowException()
             Assert.Throws<ObjectDisposedException>(() => stream.ReadFormat<byte>());
         }
 
+        [Test]
+        public void SliceStream()
+        {
+            DataStream stream = new DataStream();
+            stream.WriteByte(0xBE);
+            stream.WriteByte(0xBA);
+            stream.WriteByte(0xCA);
+            stream.WriteByte(0xFE);
+
+            Assert.That(
+                () => stream.Slice(2),
+                Throws.Nothing);
+            Assert.That(
+                () => stream.Slice(2, 2),
```

Comment:
After creating the substream we need to validate its properties `Position`, `Offset` and `Length` as well as the data it's the one we expect.